frontend/ui/widget/textviewer.lua

```diff
@@ -0,0 +1,238 @@
+local Blitbuffer = require("ffi/blitbuffer")
+local ButtonTable = require("ui/widget/buttontable")
+local CenterContainer = require("ui/widget/container/centercontainer")
+local CloseButton = require("ui/widget/closebutton")
+local Device = require("device")
+local Geom = require("ui/geometry")
+local Font = require("ui/font")
+local FrameContainer = require("ui/widget/container/framecontainer")
+local GestureRange = require("ui/gesturerange")
+local InputContainer = require("ui/widget/container/inputcontainer")
+local LineWidget = require("ui/widget/linewidget")
+local OverlapGroup = require("ui/widget/overlapgroup")
+local ScrollTextWidget = require("ui/widget/scrolltextwidget")
+local TextWidget = require("ui/widget/textwidget")
+local UIManager = require("ui/uimanager")
+local VerticalGroup = require("ui/widget/verticalgroup")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local _ = require("gettext")
+local Screen = Device.screen
+
+--[[
```

Comment:
I'm fairly sure that ldoc only picks it up if you stick this at the top of the file, although for such a paltry bit of documentation I suppose it doesn't matter. :-P